src/dawn/Optimizer/PassPrintStencilGraph.cpp

```diff
@@ -25,27 +25,26 @@ PassPrintStencilGraph::PassPrintStencilGraph() : Pass("PassPrintStencilGraph") {
 
 bool PassPrintStencilGraph::run(const std::shared_ptr<StencilInstantiation>& stencilInstantiation) {
   OptimizerContext* context = stencilInstantiation->getOptimizerContext();
-  if(context->getOptions().DumpStencilGraph) {
+  if(!context->getOptions().DumpStencilGraph || context->getOptions().Debug)
+    return true;
```

Comment:
same comment there, I would move this if DEBUG out of the pass. The if (dumpStencilGraph) was already there so we can leave it, but try to not introduce more cases of this type of pattern (with the Debug)